core/services/chainlink/application.go

```diff
@@ -708,6 +708,7 @@ func (app *ChainlinkApplication) RunJobV2(
 
 	// Some jobs are special in that they do not have a task graph.
 	if !jb.Type.HasPipelineSpec() {
+		// This is a weird situation, even if a job doesn't have a pipeline it needs a pipeline_spec_id in order to insert the run
```

Comment:
I believe after https://app.clubhouse.io/chainlinklabs/story/6065/hook-keeper-up-to-use-tasks-in-the-pipeline everything will be pipelined